src/coupling.c

```diff
@@ -99,19 +198,37 @@ void coupling_findNodeInflow(int j, double tStep)
     // --- inflow cannot drain the overland model
     if (totalCouplingInflow > 0.0)
     {
-        rawMaxInflow = (Node[j].overlandDepth * Node[j].couplingArea) / tStep;
+        rawMaxInflow = (Node_overlandDepth * Node_couplingArea) / tStep;
         maxInflow = fmin(rawMaxInflow, totalCouplingInflow);
         inflowAdjustingFactor = maxInflow / totalCouplingInflow;
-        coupling_adjustInflows(j, inflowAdjustingFactor);
+        coupling_adjustInflows(First_opening, inflowAdjustingFactor);
+
         // --- get adjusted inflows
-        opening = Node[j].coverOpening;
         while ( opening )
         {
             totalCouplingInflow += opening->newInflow;
             opening = opening->next;
         }
```

Comment:
After reflection, I am no longer convinced that this should be part of the coupling.
Various surface models might behave differently, and here we are making an assumption on the type of coupling.
For example, what if the the surface model is not 2D?
I now believe that the potential flow limiter should be carried out by the user that implement the coupling.
However, in order to keep this PR focused, this might be the topic of another, subsequent PR.

Reply:
What is exactly your suggestion here? To remove that passage of code? Remove also the coupling_adjustInflows function?

Reply:
@vtzatchkov My opinion is to remove that part of the code, and therefore `coupling_adjustInflows`. But I think it is better to do that in a subsequent commit.